ReportSimulationOutput/measure.rb

```diff
@@ -872,6 +913,76 @@ def get_outputs(runner, args)
+      next unless key == RT::Battery
+
+      resilience.variables.map { |v| v[0] }.uniq.each do |sys_id|
+        keys = resilience.variables.select { |v| v[0] == sys_id }.map { |v| v[1] }
+        vars = resilience.variables.select { |v| v[0] == sys_id }.map { |v| v[2] }
+
+        minimum_storage_state_of_charge_fraction = nil
+        batt_kwh = nil
+        batt_kw = nil
+        @hpxml.batteries.each do |battery|
+          @model.getElectricLoadCenterDistributions.each do |elcd|
+            battery_id = elcd.additionalProperties.getFeatureAsString('HPXML_ID')
+            next unless (battery_id.is_initialized && battery_id.get == battery.id)
+
+            minimum_storage_state_of_charge_fraction = elcd.minimumStorageStateofChargeFraction
+          end
+
+          batt_kw = battery.rated_power_output
+          @model.getElectricLoadCenterStorageLiIonNMCBatterys.each do |liion|
+            battery_id = liion.additionalProperties.getFeatureAsString('HPXML_ID')
+            next unless (battery_id.is_initialized && battery_id.get == battery.id)
+
+            batt_kwh = liion.additionalProperties.getFeatureAsDouble('UsableCapacity_kWh').get
+          end
+        end
+
+        if ['hourly', 'daily', 'monthly'].include?(args[:timeseries_frequency])
+          resilience_frequency = 'hourly'
+          ts_per_hr = 1
+        else
+          resilience_frequency = 'timestep'
+          ts_per_hr = @model.getTimestep.numberOfTimestepsPerHour
+        end
+
+        batt_soc = get_report_variable_data_timeseries(keys, vars, 1, 0, resilience_frequency)
+        batt_soc_kwh = batt_soc.map { |soc| soc - minimum_storage_state_of_charge_fraction }.map { |soc| soc * batt_kwh }
+        elec_prod = get_report_meter_data_timeseries(['ElectricityProduced:Facility'], UnitConversions.convert(1.0, 'J', 'kWh'), 0, resilience_frequency)
+        elec_stor = get_report_meter_data_timeseries(['ElectricStorage:ElectricityProduced'], UnitConversions.convert(1.0, 'J', 'kWh'), 0, resilience_frequency)
+        elec_prod = elec_prod.zip(elec_stor).map { |x, y| -1 * (x - y) }
+        elec = get_report_meter_data_timeseries(['Electricity:Facility'], UnitConversions.convert(1.0, 'J', 'kWh'), 0, resilience_frequency)
+        crit_load = elec.zip(elec_prod).map { |x, y| x + y }
+
+        resilience_timeseries = []
+        n_timesteps = crit_load.size
+        (0...n_timesteps).each do |init_time_step|
+          resilience_timeseries << get_resilience_timeseries(init_time_step, batt_kw, batt_soc_kwh[init_time_step], crit_load, n_timesteps, ts_per_hr)
+        end
+
+        resilience.annual_output = resilience_timeseries.sum(0.0) / resilience_timeseries.size
+        resilience.timeseries_output = resilience_timeseries
+
+        # Aggregate up from hourly to the desired timeseries frequency
+        next unless ['daily', 'monthly'].include? args[:timeseries_frequency]
+
+        year = 1999 # Try non-leap year for calculations
```

Comment:
@shorowit I could use your help here. In your emissions example, looks like you "try" a couple different years? I'm not sure how to translate that here.

Reply:
I don't remember why I use this kind of brute-force approach to figuring out if it's a leap year or not. Maybe you can come up with something better?

Reply:
Use `@hpxml.header.sim_calendar_year`?

Reply:
Yeah, maybe it's that simple...

Reply:
No wait, the issue is that you could have a leap-year specified as the calendar year but could run a weather file with 365 days, in which case you'll get 365 days of output from E+. So it's not that simple.

Reply:
Oh, looks like your use is specific to dealing with emissions hourly elec factors. For these calculations, I think we can just pass in `@hpxml.header.sim_calendar_year`.

Reply:
Hmm, `base-simcontrol-calendar-year-custom.xml` throws an error if you change the `CalendarYear` to, e.g., 2012: `Error: Specified a leap year (2012) but weather data has 8760 hours.`. Also, `hpxml.header.sim_calendar_year` gets used in other places throughout the measure...